### ISSUE TITLE
Test, improve and document the dev data population script

### ISSUE THREAD

**Issue by ragesoss** [WikiEducationFoundation/WikiEduDashboard#1468]
We  have a script that getting reasonable test data into a dev environment, by creating a few courses of different types, campaigns, etc.

It doesn't yet do the pulling of data from Wikipedia, and it isn't documented from the setup docs.

**Comment by psinghal20** [comment]
@ragesoss I think you wrote this script a while back.

**Comment by ragesoss** [comment]
True! I completely forgot about that, and I don't know that it's been put to much use. I changed task title, as it would be good to get feedback on how well it works and how it could be improved, and make sure the relevant docs point to it.

The script is `setup/populate_dashboard.rb`, and it can be run with `rake dev:populate`.

**Comment by manisha-A** [comment]
Would like to pick this up? Is this still needed?

**Comment by ragesoss** [comment]
Yes, please go for it. The idea is that running the script should be pretty quick, but should leave you with representative data to be able to access or test as much of the app's functionality as possible.

**Comment by AmitJoki** [comment]
@ragesoss can I include this in GCI? The script is already there, it needs to be tested and improved, if there's the scope for it and document the whole thing, right?

**Comment by ragesoss** [comment]
@AmitJoki yep, perfect.

**Comment by urbanecm** [comment]
Hello, that's me again :-). What is "improve"? Tasks in GCI need to have clear criteria, so students know what is expected from them and what is and is not acceptable. I'd maybe split the task into several smaller tasks, such as "make the script do X", then "make the script do Y" and "document the script". All of those tasks can be done by different students - it doesn't have to be one large task.

**Comment by AmitJoki** [comment]
@ragesoss @bwreid per @urbanecm 's comments, the generic word "improve" needs to be elaborated more so we can quantify the work done. Do you have any particular improvement in mind that could be done or is it optional?

**Comment by urbanecm** [comment]
Ping? :-)

**Comment by AmitJoki** [comment]
@ragesoss I will reword this task to just test the existing script and if we have a more objective idea of how to improve the same, I'll add another task for the same. Is that okay @urbanecm?

**Comment by urbanecm** [comment]
Perfect for me :)

**Comment by AmitJoki** [comment]
@urbanecm done :)

**Comment by urbanecm** [comment]
Thanks! A question, which submission should the students submit into the repo?

**Comment by AmitJoki** [comment]
The tests and documentation can be submitted as a pull request to this repository. @urbanecm

**Comment by Ajitesh13** [comment]
Hello, this issue is still open. Can you please specify what need to be done here, as I am a new contributor, willing to contribute in this repo.

**Comment by ragesoss** [comment]
@Ajitesh13 for this issue, you should run the data population script (`rake dev:populate`), report any errors, and then explore the courses it added to the Dashboard and then document which features of the system that you would need to add more/different data in order to test locally.

**Comment by Ajitesh13** [comment]
Thanks @ragesoss for the insights, I will let you know, if I faced any difficulty.

**Comment by Ajitesh13** [comment]
Where will I write the documentation to open a PR ?

**Comment by Ajitesh13** [comment]
@ragesoss Where will I write the documentation to open a PR? Is it in docs/user_roles.md#populating-initial-courses

**Comment by ragesoss** [comment]
@Ajitesh13 the documentation for opening a PR is in the CONTRIBUTING file: https://github.com/WikiEducationFoundation/WikiEduDashboard/blob/master/CONTRIBUTING.md#pull-request-process

**Comment by varu567** [comment]
hey @ragesoss would like to contribute to it.

**Comment by ragesoss** [comment]
@varu567 cool, go for it. let us know you have any questions.

**Comment by Nikuunj** [comment]
Hi @ragesoss,

I hope this message finds you well. I'm interested in joining the WikiEducationFoundation  Slack channel to stay informed and contribute to the discussions. Could you please add me to the channel?

Looking forward to being a part of the community.

**Comment by Nikuunj** [comment]
@ragesoss   email - mnikunj262@gmail.com

**Comment by Shrutikaghule** [comment]
Hi @ragesoss 
is the issue still open
Id like to solve it also  I m interested in joining the WikiEducationFoundation  Slack so to contribute further.
email-ghuleshrutika20@gmail.com

**Comment by ragesoss** [comment]
@Shrutikaghule yes, this is one that always has room for further testing and improvement. I sent invite ... I assume "email-" is not part of your email address.

**Comment by mustalidhanerawala** [comment]
Hello @ragesoss 
My name is Mustali, I am a newcomer to open source, I have always used Wikipedia for my school projects and quizzes. Now when I am entering the world of open source contributing I want to begin from here, Can you help me where can I start? Or what issue can I solve? Also can add me in your Slack account. Here is my email:
dhanerawala5@hotmail.com
Thank You.

**Comment by ragesoss** [comment]
@mustalidhanerawala i've sent a Slack invite. This is the best entry point for figuring out how to get started: https://github.com/WikiEducationFoundation/WikiEduDashboard/blob/master/CONTRIBUTING.md

**Comment by naveensrinivas282** [comment]
Hello @ragesoss ,
Is this issue still open ? I would like to contribute to this issue if possible
Also is there any progress that I should build on or I should fix all issues I find ?

**Comment by ragesoss** [comment]
> Hello [@ragesoss](https://github.com/ragesoss) , Is this issue still open ? I would like to contribute to this issue if possible Also is there any progress that I should build on or I should fix all issues I find ?

it's still open. the issue is a little vague, but if you understand it well enough to dive in, go for it.

**Comment by naveensrinivas282** [comment]
I just need to test out the current script and report any errors I find. See if Dashboard is getting tested properly and fix if there is any issue. Document the script in `setup` if needed

Is that all or did I miss something @ragesoss ?

**Comment by ragesoss** [comment]
@naveen28204280 the 'improve' aspect of this includes expanding the script to load a reasonable selection of additional data, to create a more complete small-scale representative database state that covers the full range of data present in a production database.

**Comment by NIGHTFURY609** [comment]
hey @ragesoss  can i take up this issue?

**Comment by ragesoss** [comment]
> hey [@ragesoss](https://github.com/ragesoss) can i take up this issue?

yes! (Apologies for the delayed reply; I'm just catching up after two weeks of travel.)
